pre_commit/languages/lua.py

```diff
@@ -0,0 +1,113 @@
+import os
+import re
+from typing import Generator
+from typing import Sequence
+from typing import Tuple
+
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.envcontext import Var
+from pre_commit.hook import Hook
+from pre_commit.languages import helpers
+from pre_commit.prefix import Prefix
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import cmd_output
+
+ENVIRONMENT_DIR = 'lua_env'
+get_default_version = helpers.basic_get_default_version
+healthy = helpers.basic_healthy
+
+
+def install_environment(
+    prefix: Prefix,
+    version: str,
+    additional_dependencies: Sequence[str],
+) -> None:
+    helpers.assert_version_default('lua', version)
+    # luarocks, Lua's package manager, is separate from the runtime.
+    helpers.assert_version_default('luarocks', version)
+
+    envdir = _envdir(prefix, version)
+    with clean_path_on_failure(envdir):
+        with in_env(prefix, version):
+            _bootstrap_environment(envdir)
+
+            # Sometimes a rockspec is not available (like with a local hook).
+            # Check before trying to build.
+            root_files = os.listdir(prefix.prefix_dir)
+            if any(filename.endswith('.rockspec') for filename in root_files):
+                build_cmd = ('luarocks', '--tree', envdir, 'build')
+                helpers.run_setup_cmd(prefix, build_cmd)
+
+            # luarocks can't install multiple packages at once
+            # so install them individually.
+            for dependency in additional_dependencies:
+                cmd = ('luarocks', '--tree', envdir, 'install', dependency)
+                helpers.run_setup_cmd(prefix, cmd)
+
+
+def run_hook(
+    hook: Hook,
+    file_args: Sequence[str],
+    color: bool,
+) -> Tuple[int, bytes]:
+    with in_env(hook.prefix, hook.language_version):
+        return helpers.run_xargs(hook, hook.cmd, file_args, color=color)
+
+
+@contextlib.contextmanager
+def in_env(
+        prefix: Prefix,
+        language_version: str,
+) -> Generator[None, None, None]:
+    with envcontext(get_env_patch(_envdir(prefix, language_version))):
+        yield
+
+
+def get_env_patch(env: str) -> PatchesT:
+    version = _get_lua_path_version()
+    return (
+        ('PATH', (os.path.join(env, 'bin'), os.pathsep, Var('PATH'))),
+        (
+            'LUA_PATH', (
+                os.path.join(env, 'share', 'lua', version, '?.lua;'),
+                os.path.join(env, 'share', 'lua', version, '?', 'init.lua;'),
+                Var('LUA_PATH'),
+            ),
+        ),
+        (
+            'LUA_CPATH', (
+                os.path.join(env, 'lib', 'lua', version, '?.so;'),
+                Var('LUA_CPATH'),
+            ),
+        ),
+    )
+
+
+def _get_lua_path_version() -> str:
+    """Get the Lua version used in file paths."""
+    returncode, stdout, _ = cmd_output('lua', '-e', 'print(package.path)')
+    if returncode != 0:
+        return 'unknown'
```

Comment:
this seems kind of fragile -- I think it would be better to raise an error if it can't be detected at all?